python/pyarrow/tests/conftest.py

```diff
@@ -15,15 +15,30 @@
 # specific language governing permissions and limitations
 # under the License.
 
+import os
 import pytest
+import hypothesis as h
 
 try:
     import pathlib
 except ImportError:
     import pathlib2 as pathlib  # py2 compat
 
 
+# setup hypothesis profiles
+h.settings.register_profile('ci', max_examples=1000)
+h.settings.register_profile('dev', max_examples=10)
+h.settings.register_profile('debug', max_examples=10,
+                            verbosity=h.Verbosity.verbose)
```

Comment:
@wesm We can also use profiles. I have not enabled hypothesis on the CI services yet. Without setting a seed it can be indeterministic.

Reply:
I think it's ok for `hypothesis` to be indeterministic as it will print out the failing example. Then we can use this reproduce the failing test locally.